src/main/java/com/github/manolo8/darkbot/core/manager/HangarManager.java

```diff
@@ -0,0 +1,222 @@
+
+                    if (active) {
+                        JsonArray dronesArray = hangar.getAsJsonObject().get("general").getAsJsonObject().get("drones").getAsJsonArray();
+                        for (JsonElement dron : dronesArray){
+                            JsonObject dronJson = dron.getAsJsonObject();
+                            String lootId = "drone_iris";
+                            switch (dronJson.get("L").getAsInt()){
+                                case 1:
+                                    lootId = "drone_flax";
+                                    break;
+                                case 2:
+                                    lootId = "drone_iris";
+                                    break;
+                                case 3:
+                                    lootId = "drone_apis";
+                                    break;
+                                case 4:
+                                    lootId = "drone_zeus";
+                                    break;
+                            }
+                            int damage = Integer.parseInt(dronJson.get("HP").getAsString().replace("%"," ").trim());
+                            this.drones.add(new Dron(lootId,dronJson.get("repair").getAsInt(),dronJson.get("I").getAsString(),
+                                    dronJson.get("currency").getAsString(),dronJson.get("LV").getAsInt(),damage));
+                        }
+                    }
+                }
+            }
+
+        } catch (Exception e){}
+    }
+
+    private boolean repairDron(Dron dron, String activeHangarId){
+        try {
+            String decodeParams =
+                    "{\"action\":\"repairDrone\",\"lootId\":\"" + dron.getLootId() + "\",\"repairPrice\":" + dron.getRepairPrice() +
+                            ",\"params\":{\"hi\":" + activeHangarId + "}," +
+                            "\"itemId\":\"" + dron.getItemId() + "\",\"repairCurrency\":\"" +dron.getRepairCurrency() +
+                            "\",\"quantity\":1,\"droneLevel\":" + dron.getDroneLevel() + "}";
+            String encodeParams = Base64.getEncoder().encodeToString(decodeParams.getBytes("UTF-8"));
+            String url = "/flashAPI/inventory.php?action=repairDrone&params="+encodeParams;
+            String json = getDataInventory(url);
+            if (json.contains("'isError':0")){
+                return true;
+            } else {
+                return false;
+            }
+        } catch (Exception e){
+            return false;
+        }
+    }
+
+    public void updateHangars() {
+        String params = "/flashAPI/inventory.php?action=getHangarList";
+        String decodeString = getDataInventory(params);
+
+        JsonObject object =  new JsonParser().parse(decodeString).getAsJsonObject();
+        JsonArray hangarsArray  = object.get("data").getAsJsonObject().get("ret").getAsJsonObject().get("hangars").getAsJsonArray();
```

Comment:
Why don't you map this directly to the java object?

Reply:
The object to make the deserialization in a fast way is much bigger and I only need two data.